utils/swift_build_support/swift_build_support/products/swiftformat.py

```diff
@@ -0,0 +1,104 @@
+# swift_build_support/products/swiftformat.py -----------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+import platform
+
+from build_swift.build_swift.constants import MULTIROOT_DATA_FILE_PATH
+
+from . import cmark
+from . import foundation
+from . import libcxx
+from . import libdispatch
+from . import libicu
+from . import llbuild
+from . import llvm
+from . import product
+from . import swift
+from . import swiftpm
+from . import swiftsyntax
+from . import xctest
+from .. import shell
+
+
+class SwiftFormat(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        """product_source_name() -> str
```

Comment:
> are doc strings inherited from super classes?

It depends on the version of Python you're using. Inheritance for docstrings was added in Python 3 (I think 3.5, but can't recall or find a reference quickly). I believe this project uses Python 2.X so doc strings aren't inherited in this context.

For additional context, note that the other `Product` subclasses duplicate this doc string as well.

Let me know if you would prefer that I delete the doc string or otherwise refer to the superclass' doc string. If so, you may want to apply the same to the other `Product` subclasses.

Reply:
>  note that the other Product subclasses duplicate this doc string as well.

Yeah, it looks like we're inconsistent about this in two different ways:
* Not all products duplicate doc strings
* For products that do, they are not duplicating the doc strings for all methods, just some (e.g. this one).

Anyway, I won't hold up the patch over this - we can clean it up separately across all products.

Reply:
I agree there's inconsistency here. I'm fine with whatever approach you prefer. I don't write Python very often so I can't claim to know what's the best practice here.